src/main/python/mainQt.py

```diff
@@ -225,10 +225,11 @@ def reset_ui_record_number(self):
         self.ui.horizontalSlider.setValue(0)
 
     def open_dialog(self) -> None:
-        """
-        TODO : To be implemented
-        """
-        pass
+        fileName, _ = QFileDialog.getOpenFileName(
+            self, "Open record", "", "Sim Replay files (*.sr);;All files (*.*)"
+        )
+        if fileName:
+            self._model.open_file(fileName)
 
     def open_import_window(self) -> None:
         self._import_window = ImportWindow(self._model, parent=self, f=Qt.WindowType.Dialog)
```

Comment:
### Major issue :

import is processed without the `MainTableModel` object is cleared, resulting on an append instead of overwrite 
like 'open' action, we shall have :
```python
self._mainTableModel.clear()
```